main.go

```diff
@@ -143,28 +144,58 @@ func registerRHSM(ctx *cli.Context) (error, string) {
 				ctx.StringSlice("activation-key"),
 				ctx.String("server"))
 		} else {
+			var orgs []string
 			if organization != "" {
-				err = registerPassword(username, password, organization, ctx.String("server"))
+				_, err = registerUsernamePassword(username, password, organization, ctx.String("server"))
 			} else {
-				/* TODO: When organization was not specified using CLI option --organization and it is
+				orgs, err = registerUsernamePassword(username, password, "", ctx.String("server"))
+				/* When organization was not specified using CLI option --organization, and it is
 				   required, because user is member of more than one organization, then ask for
-				   the organization. RHSM D-Bus API can provide list of available organizations
-				   in this case. */
-				err = registerPassword(username, password, "", ctx.String("server"))
+				   the organization. */
+				if len(orgs) > 0 {
+					// Stop spinner to be able to display message and ask for organization
+					if isColorful {
+						s.Stop()
+					}
+
+					// Ask for organization and display hint with list of organizations
+					scanner := bufio.NewScanner(os.Stdin)
+					fmt.Println("Available Organizations:")
+					writer := tabwriter.NewWriter(os.Stdout, 0, 2, 2, ' ', 0)
+					for i, org := range orgs {
+						_, _ = fmt.Fprintf(writer, "%v\t", org)
+						if (i+1)%4 == 0 {
+							_, _ = fmt.Fprint(writer, "\n")
+						}
+					}
+					_ = writer.Flush()
+					fmt.Print("\nOrganization: ")
+					_ = scanner.Scan()
```

Comment:
It is good point, but it is generic problem of `rhc`. When you try to register using `rhc` and you redirect output to e.g. some file like this `rhc connect > ./log.txt` (no username/password), then `rhc` blocks without any error written to `stderr`. I think we could create issue for this. I think that this could be something nice, useful and simple to fix for Go newbie. ;-)